src/stdlib/non-det.prelude.ts

```diff
@@ -0,0 +1,28 @@
+export const nonDetPrelude = `
+    function require(predicate) {
+        return predicate ? "Satisfied require" : amb();
+    }
+
+    /* DISTINCT */
+    /* The distinct function checks whether the items in a list are unique. */
+    /* Taken from SICP JS section 4.3.2 */
+
+    function distinct(items) {
```

Comment:
The function distinct is not needed in the prelude of non-det. It can be easily added in each example that needs it.